### PR TITLE
Bandit security recommendations for Cylc 8 target

### PR DESCRIPTION
**Pull request by MartinRyan**
recreation of  pull request for the new Cylc 8

### CODE REVIEW

**Comment by hjoliver** [comment]
@matthewrmshin - can you review or reassign at the UK end?

**Comment by hjoliver** [comment]
Sorry @MartinRyan this is now conflicted too!  At this point we owe you some help with the conflicts if they are non-trivial.  @matthewrmshin - can we try to get this PRs done as soon as possible, Martin has suffered long enough!

**Comment by matthewrmshin** [comment]
(OK. On the case. This will be my top priority of the day for Cylc.)

File: lib/cylc/tests/test_cylc_subproc.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import unittest
+
+from cylc.cylc_subproc import procopen
+from mock import call
```
Comment (matthewrmshin) [inline lib/cylc/tests/test_cylc_subproc.py]: Can now do `from unittest.mock ...`.

File: .travis/install.sh [hunk 2]
```diff
@@ -33,7 +33,7 @@ if grep -E '(unit-tests|functional-tests)' <<< "${args[@]}"; then
 fi
 
 if grep 'unit-tests' <<< "${args[@]}"; then
-    pip install pycodestyle pytest
+    pip install pycodestyle pytest mock testfixtures
```
Comment (matthewrmshin) [inline .travis/install.sh]: You can now use `unittest.mock` from standard library.

File: lib/cylc/tests/test_cylc_subproc.py (hunk 1 above)
Comment (hjoliver) [reply lib/cylc/tests/test_cylc_subproc.py]: Done.

File: .travis/install.sh (hunk 2 above)
Comment (hjoliver) [reply .travis/install.sh]: Done.

**Comment by hjoliver** [comment]
@matthewrmshin - I've resolved the conflicts and pushed to @MartinRyan's branch. Should be good to merge if tests pass again.

**Comment by hjoliver** [comment]
(and addressed your comments ... except for the "future PR" one).

**Pull request merged**